@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,255 @@
+import AddIcon from '@mui/icons-material/Add'
+import React from 'react'
+import SettingsBackupRestoreIcon from '@mui/icons-material/SettingsBackupRestore'
+import { Map } from 'immutable'
+import { SelectChangeEvent } from '@mui/material'
+import { withState } from 'reaclette'
+
+import ActionButton from './ActionButton'
+import Button from './Button'
+import Checkbox from './Checkbox'
+import Input from './Input'
+import IntlMessage from './IntlMessage'
+import Select from './Select'
+import { alert } from './Modal'
+
+import XapiConnection, { ObjectsByType, Pif, PifMetrics } from '../libs/xapi'
+
+interface ParentState {
+  objectsByType: ObjectsByType
+  objectsFetched: boolean
+  xapi: XapiConnection
+}
+
+interface State {
+  form: {
+    [key: string]: unknown
+    bondMode: string
+    description: string
+    isBonded: boolean
+    isEmptyLabel: boolean
+    mtu: string
+    nameLabel: string
+    pifsId: string | string[]
+    vlan: string
+  }
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  createNetwork: () => Promise<void>
+  handleChange: (e: SelectChangeEvent<unknown> | React.ChangeEvent<{ name: string; value: unknown }>) => void
+  resetForm: () => void
+  toggleBonded: () => void
+}
+
+interface Computed {
+  collection?: Pif[]
+  pifs?: Map<string, Pif>
+  pifsMetrics?: Map<string, PifMetrics>
+}
+
+const BOND_MODE = ['active-backup', 'balance-slb', 'lacp']
+
+const BUTTON_STYLES = {
+  marginRight: '1em',
+  width: 'fit-content',
+}
+
+const OPTION_PIF_RENDERER = (pif: Pif, { pifsMetrics }: { pifsMetrics: Computed['pifsMetrics'] }) =>
+  `${pif.device} (${pifsMetrics?.find(metrics => metrics.$ref === pif.metrics)?.device_name ?? 'unknown'})`
+
+const INPUT_STYLES = {
+  marginBottom: '1em',
+  width: '100%',
+}
+
+const getInitialFormState = (): State['form'] => ({
+  bondMode: '',
+  description: '',
+  isBonded: false,
+  isEmptyLabel: false,
+  mtu: '',
+  nameLabel: '',
+  pifsId: '',
+  vlan: '',
+})
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      form: getInitialFormState(),
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
```

Comment:
We are using this computed only in the [collection computed](https://github.com/vatesfr/xen-orchestra/blob/xol-networkModification/%40xen-orchestra/lite/src/components/AddNetwork.tsx#L90). Should we have to calculate it inside the `collection` ?

Reply:
If I'm not mistaken, this will be equivalent in terms of performance since when a PIF changes, it will trigger both computed no matter what, so yes, we can get rid of one computed here.